chain/network/src/types.rs

```diff
@@ -511,6 +513,23 @@ pub enum NetworkClientResponses {
     Ban { ban_reason: ReasonForBan },
 }
 
+/// Wraps `NetworkClientMessages` attaching an opentelemetry Context to it.
+/// Lets us trace async work across different actix actors.
+#[derive(actix::Message, Debug)]
+#[rtype(result = "NetworkClientResponses")]
+pub struct NetworkClientMessagesWithContext {
+    pub msg: NetworkClientMessages,
+    pub context: opentelemetry::Context,
+}
+
```

Comment:
I wonder if it would make sense to make this generic, that is,

```rust
#[derive(actix::Message, Debug)]
#[rtype(result = "<T as actix::Message>::Result")]
pub struct WithContext<T> {
    pub msg: T,
    pub context: opentelemetry::Context,
}
```

or something. Do `NetworkClientMessages` encapsulate all of the interesting cross-actor work we have?

Reply:
I also wonder if it would have been a good idea to add a method to the `MsgRecipient` trait to “automatically” contextualize the message without the caller code needing to deal with this type directly.

Reply:
Please take a look at #7773

I'm not sure what you mean by `add a method to the MsgRecipient trait`. The context needs to be attached when the `actix::Message` is created, not when a Recipient picks it up.